### PR TITLE
fix(mistral, pism): Allows PISM to compile on mistral

### PR DESCRIPTION
**Pull request by pgierz**

### CODE REVIEW

**Review by dbarbi — COMMENTED** [review]
Are there GCC_NAME settings for other machines yet?

**Comment by pgierz** [comment]
No, not yet. That would have bitten me in the butt after lunch when I try to install pism on ollie....

**Comment by dbarbi** [comment]
Do these changes break comething for ollie / hlrn users then?

File: configs/machines/mistral.yaml [hunk 1]
```diff
@@ -109,7 +109,7 @@ export_vars:
         # NOTE(PG): Is it intentional to have different netcdf_fortran and
         # netcdf_c versions?
         - "NETCDFFROOT=/sw/rhel6-x64/netcdf/netcdf_fortran-4.4.2-parallel-impi-intel14/"
-        - "NETCDFROOT=/sw/rhel6-x64/netcdf/netcdf_c-4.3.2-parallel-impi-intel14/"
+        - "NETCDFROOT=/sw/rhel6-x64/netcdf/netcdf_c-4.4.1.1-parallel-impi-intel14/"
```
Comment (seb-wahl) [inline configs/machines/mistral.yaml:112]: I don't know whether FOCI likes those changes. I have to test first

File: configs/machines/mistral.yaml [hunk 2]
```diff
@@ -152,7 +152,7 @@ export_vars:
         #
         - "LD_LIBRARY_PATH=$NETCDFROOT/lib:$NETCDFFROOT/lib:$HDF5ROOT/lib:$LD_LIBRARY_PATH"
         # avoid GLIBCXX_3.4.15 not found error
-        - "LD_LIBRARY_PATH=/sw/rhel6-x64/gcc/gcc-4.8.2/lib64:$LD_LIBRARY_PATH"
+        - "LD_LIBRARY_PATH=/sw/rhel6-x64/gcc/gcc-6.4.0/lib64:$LD_LIBRARY_PATH"
```
Comment (seb-wahl) [inline configs/machines/mistral.yaml:155]: same as above

**Comment by seb-wahl** [comment]
Tried to test it but esm-tools don't work anymore on mistral for FOCI: (fine on HLRN), same if I'm on release or the `hotfix/pism_mistral` branch:
```
b350071@mlogin103% esm_versions check
+---------------------+-----------+---------------------------------------------------------------+----------+---------+
| package_name        | version   | file                                                          | branch   | tags    |
|---------------------+-----------+---------------------------------------------------------------+----------+---------|
| esm_calendar        | 5.0.0     | /mnt/lustre01/pf/b/b350071/.local/lib/python3.6/site-packages |          |         |
| esm_database        | 5.0.0     | /mnt/lustre01/pf/b/b350071/.local/lib/python3.6/site-packages |          |         |
| esm_environment     | 5.0.0     | /mnt/lustre01/pf/b/b350071/.local/lib/python3.6/site-packages |          |         |
| esm_master          | 5.0.1     | /mnt/lustre01/pf/b/b350071/.local/lib/python3.6/site-packages |          |         |
| esm_motd            | 5.0.2     | /mnt/lustre01/pf/b/b350071/.local/lib/python3.6/site-packages |          |         |
| esm_parser          | 5.0.4     | /mnt/lustre01/pf/b/b350071/.local/lib/python3.6/site-packages |          |         |
| esm_plugin_manager  | 5.0.0     | /mnt/lustre01/pf/b/b350071/.local/lib/python3.6/site-packages |          |         |
| esm_profile         | 5.0.0     | /mnt/lustre01/pf/b/b350071/.local/lib/python3.6/site-packages |          |         |
| esm_rcfile          | 5.0.0     | /mnt/lustre01/pf/b/b350071/.local/lib/python3.6/site-packages |          |         |
| esm_runscripts      | 5.0.14    | /mnt/lustre01/pf/b/b350071/.local/lib/python3.6/site-packages |          |         |
| esm_tools           | 5.0.13    | /mnt/lustre02/work/bb0519/b350071/esm/esm_tools               | release  | v5.0.13 |
| esm_version_checker | 5.1.2     | /mnt/lustre01/pf/b/b350071/.local/lib/python3.6/site-packages |          |         |
+---------------------+-----------+---------------------------------------------------------------+----------+---------+

b350071@mlogin103% esm_master get-foci-default -c
************************************************************************************
Message found for package esm_tools version 5.0.13:

"ESM Tools version 5.0.5 allows for the generation of hosing files in AWI-ESM-2.1 and AWI-ESM-2.2
To use that, you still need to install a plugin. Please run:

$ pip install --user gfw_creator"

Upgrade this package by typing:              esm_versions upgrade esm_tools
Upgrade all packages by typing:              esm_versions upgrade
************************************************************************************
 steps_since_initial
Traceback (most recent call last):
  File "/pf/b/b350071/.local/bin/esm_master", line 8, in <module>
    sys.exit(main())
  File "/pf/b/b350071/.local/lib/python3.6/site-packages/esm_master/cli.py", line 76, in main
    main_flow(parsed_args, target)
  File "/pf/b/b350071/.local/lib/python3.6/site-packages/esm_master/esm_master.py", line 36, in main_flow
    complete_setup = SimulationSetup(user_config=user_config)
  File "/pf/b/b350071/.local/lib/python3.6/site-packages/esm_runscripts/sim_objects.py", line 42, in __init__
    self.config = prepare.run_job(self.config)
  File "/pf/b/b350071/.local/lib/python3.6/site-packages/esm_runscripts/prepare.py", line 5, in run_job
    helpers.evaluate(config, "prepare", "prepare_recipe")
  File "/pf/b/b350071/.local/lib/python3.6/site-packages/esm_runscripts/helpers.py", line 69, in evaluate
    framework_recipe, framework_plugins, config
  File "/pf/b/b350071/.local/lib/python3.6/site-packages/esm_plugin_manager/esm_plugin_manager.py", line 130, in work_through_recipe
    config = getattr(submodule, workitem)(config)
  File "/pf/b/b350071/.local/lib/python3.6/site-packages/esm_runscripts/prepare.py", line 604, in finalize_config
    config.finalize()
  File "/pf/b/b350071/.local/lib/python3.6/site-packages/esm_parser/esm_parser.py", line 2637, in finalize
    self.run_recursive_functions(self)
  File "/pf/b/b350071/.local/lib/python3.6/site-packages/esm_parser/esm_parser.py", line 2644, in run_recursive_functions
    recursive_run_function([], config, "atomic", perform_actions, config)
  File "/pf/b/b350071/.local/lib/python3.6/site-packages/esm_parser/esm_parser.py", line 1553, in recursive_run_function
    tree + [key], value, level, func, *args, **kwargs
  File "/pf/b/b350071/.local/lib/python3.6/site-packages/esm_parser/esm_parser.py", line 1553, in recursive_run_function
    tree + [key], value, level, func, *args, **kwargs
  File "/pf/b/b350071/.local/lib/python3.6/site-packages/esm_parser/esm_parser.py", line 1526, in recursive_run_function
    right = func(tree + [None], right, *args, **kwargs)
  File "/pf/b/b350071/.local/lib/python3.6/site-packages/esm_parser/esm_parser.py", line 2129, in perform_actions
    source = int(source)
ValueError: invalid literal for int() with base 10: ' steps_since_initial'
```
Seems to be related to the comment on slack (where no answer was given).
Any ideas what's wrong. Seems that some variables are not resolved correctly anymore. `seconds_since_initial` is used on nemo.yaml and oifs.yaml

**Comment by pgierz** [comment]
Alright, if GXX_NAME and GCC_NAME aren't set, CMake just hangs. Not sure what to put there for ollie though....

**Comment by pgierz** [comment]
Ok I just had another look. Unless I forgot a pull somewhere, PISM builds on Ollie, not on mistrial though. Runtime issues may still be there. In particular one thing we will need to look out for is running both models from the same runscript.

@dbarbi, you looked at my parallel srun stuff. I would try to get pism to build on both machines tomorrow. Any nasty pitfalls I should keep my eye on? PISM works but then echam merps? Or the other way?

**Comment by mandresm** [comment]
@pgierz , was this merged in some way and can it be closed or does it still need to be considered?

**Comment by pgierz** [comment]
This looks ancient. I will close for right now and if needed make a new issue the next time something pops up on mistral, but I have not had any users complain recently...

**Pull request closed without merging**